src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,162 @@
+        afterNode.prev = newNode;
+        if (beforeNode != null) {
+            beforeNode.next = newNode;
+        } else {
+            firstNode = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element: list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNode(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> operatedNode = getNode(index);
+        T oldValue = operatedNode.item;
+        operatedNode.item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> removed = getNode(index);
+        T value = removed.item;
+        unlink(removed);
+        return value;
+
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        Node<T> node = getNode(object);
+        if (node == null) {
+            return false;
+        }
+        unlink(node);
+        return true;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private Node<T> getNode(int index) {
+        checkIndex(index);
+        Node<T> toSearch;
+        if (index < size / 2) {
+            toSearch = firstNode;
+            for (int i = 0; i < index; i++) {
+                toSearch = toSearch.next;
+            }
+        } else {
+            toSearch = lastNode;
+            for (int i = size - 1; i > index; i--) {
+                toSearch = toSearch.prev;
+            }
+        }
+        return toSearch;
+    }
+
+    private Node<T> getNode(T value) {
+        Node<T> node = firstNode;
+        for (int i = 0; i < size; i++) {
+            if ((node.item == value) || (node.item != null && node.item.equals(value))) {
+                return node;
+            }
+            node = node.next;
+        }
+        return null;
+    }
+
+    private void unlink(Node<T> node) {
+        Node<T> prev = node.prev;
+        Node<T> next = node.next;
+        if (next == null) {
+            if (prev != null) {
+                prev.next = null;
+            }
+            lastNode = prev;
+        } else if (prev == null) {
+            next.prev = null;
+            firstNode = next;
+        } else {
+            next.prev = prev;
+            prev.next = next;
+        }
+        size--;
+    }
+
+    private void checkIndex(int index) {
+        if (index < 0) {
+            throw new IndexOutOfBoundsException("Can't operate negative index");
+        } else if (index >= size) {
+            throw new IndexOutOfBoundsException("Index " + index + " ot of bounds of size " + size);
+        }
```

Comment:
it is better to combine these checks and raise an error with the inscription "Index " + index + " must be in the range from 0 to " + size; . Or something like that